src/key_info_managers/mod.rs

```diff
@@ -96,14 +116,14 @@ trait ManageKeyInfo {
     /// # Errors
     ///
     /// Returns an error as a String if there was a problem accessing the Key Info Manager.
-    fn get(&self, key_triple: &KeyTriple) -> Result<Option<&KeyInfo>, String>;
+    fn get(&self, key_identity: &KeyIdentity) -> Result<Option<&KeyInfo>, String>;
 
     /// Returns a Vec of reference to the key triples corresponding to this provider.
     ///
     /// # Errors
     ///
     /// Returns an error as a String if there was a problem accessing the Key Info Manager.
-    fn get_all(&self, provider_id: ProviderId) -> Result<Vec<&KeyTriple>, String>;
+    fn get_all(&self, provider_identity: ProviderIdentity) -> Result<Vec<KeyIdentity>, String>;
```

Comment:
I think passing a reference would be better here. But maybe `provider_name: &str` is enough?

Reply:
Chose to do it like this as it will need to follow the common API of `ManageKeyInfo`.`get_all` method, which will later be used by the `SQLiteKeyInfoManager`. And as the `SQLiteKeyInfoManager` will use this. As to whether we're just using `provider_name`s to distinguish providers is another topic. I personally think provider names should be unique 🤷🏼‍♂️